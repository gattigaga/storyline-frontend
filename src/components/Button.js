import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const Content = styled.button`
  padding: 12px 16px;
  background: ${({ isOutlined }) => (isOutlined ? "white" : "#e74c3c")};
  color: ${({ isOutlined }) => (isOutlined ? "#e74c3c" : "white")};
  border: 1px solid ${({ isOutlined }) => (isOutlined ? "#e74c3c" : "white")};
  font-family: Roboto;
  font-size: 14px;
  cursor: pointer;
  border-radius: 4px;
  outline: none;
`;

const Button = ({ className, caption, isOutlined, onClick }) => (
  <Content className={className} isOutlined={isOutlined} onClick={onClick}>
    {caption}
  </Content>
);

Button.propTypes = {
  caption: PropTypes.string,
  isOutlined: PropTypes.bool,
  className: PropTypes.string,
  onClick: PropTypes.func
};

Button.defaultProps = {
  caption: "Caption"
};

export default Button;
