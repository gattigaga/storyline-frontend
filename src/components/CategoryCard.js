import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { Link } from "react-router-dom";

import { separateByComma } from "../helpers/formatter";
import Button from "../components/Button";

const StyledLink = styled(Link)`
  text-decoration: none;
`;

const Container = styled.div`
  flex: 1;
  padding: 24px;
  border: 1px solid #ccc;
  box-sizing: border-box;
  cursor: pointer;
`;

const Label = styled.h1`
  font-family: Roboto;
  font-size: 32px;
  color: #333;
  margin-top: 0px;
  margin-bottom: 8px;
`;

const Total = styled.p`
  font-family: Roboto;
  font-size: 16px;
  color: #aaa;
  margin-top: 0px;
  margin-bottom: 32px;
`;

const CategoryCard = ({
  label,
  totalStories,
  href,
  isChosen,
  withoutButton,
  onClickChoose
}) => (
  <StyledLink to={href}>
    <Container>
      <Label>{label}</Label>
      <Total>{separateByComma(totalStories)} stories</Total>
      {!withoutButton && (
        <Button
          caption={isChosen ? "Chosen" : "Choose"}
          onClick={event => {
            event.preventDefault();
            onClickChoose();
          }}
          isOutlined={!isChosen}
        />
      )}
    </Container>
  </StyledLink>
);

CategoryCard.propTypes = {
  label: PropTypes.string,
  totalStories: PropTypes.number,
  href: PropTypes.string,
  isChosen: PropTypes.bool,
  withoutButton: PropTypes.bool,
  onClickChoose: PropTypes.func
};

export default CategoryCard;
