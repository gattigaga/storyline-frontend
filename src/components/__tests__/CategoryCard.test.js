import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import CategoryCard from "../CategoryCard";
import Button from "../Button";

describe("CategoryCard", () => {
  const setup = propOverrides => {
    const props = {
      label: "Technology",
      totalStories: 15326,
      href: "/category/technology",
      onClickChoose: jest.fn(),
      ...propOverrides
    };

    const wrapper = shallow(<CategoryCard {...props} />);

    return {
      props,
      wrapper
    };
  };

  it("should renders default", () => {
    const { wrapper } = setup();

    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("should renders without button", () => {
    const { wrapper } = setup({ withoutButton: true });

    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("should renders as chosen", () => {
    const { wrapper } = setup({ isChosen: true });

    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("should call 'onClickChoose' callback when choose button pressed", () => {
    const { wrapper, props } = setup();
    const event = {
      preventDefault: () => {}
    };

    wrapper.find(Button).simulate("click", event);

    expect(props.onClickChoose).toBeCalled();
  });
});
