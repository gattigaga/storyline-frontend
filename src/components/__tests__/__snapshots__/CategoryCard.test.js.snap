// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryCard should renders as chosen 1`] = `
<Styled(Link)
  to="/category/technology"
>
  <styled.div>
    <styled.h1>
      Technology
    </styled.h1>
    <styled.p>
      15,326
       stories
    </styled.p>
    <Button
      caption="Chosen"
      isOutlined={false}
      onClick={[Function]}
    />
  </styled.div>
</Styled(Link)>
`;

exports[`CategoryCard should renders default 1`] = `
<Styled(Link)
  to="/category/technology"
>
  <styled.div>
    <styled.h1>
      Technology
    </styled.h1>
    <styled.p>
      15,326
       stories
    </styled.p>
    <Button
      caption="Choose"
      isOutlined={true}
      onClick={[Function]}
    />
  </styled.div>
</Styled(Link)>
`;

exports[`CategoryCard should renders without button 1`] = `
<Styled(Link)
  to="/category/technology"
>
  <styled.div>
    <styled.h1>
      Technology
    </styled.h1>
    <styled.p>
      15,326
       stories
    </styled.p>
  </styled.div>
</Styled(Link)>
`;
